src/main/java/com/github/alexthe666/iceandfire/entity/props/SirenData.java

```diff
@@ -0,0 +1,200 @@
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.Level;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.UUID;
+
+
+public class SirenData {
+    public @Nullable EntitySiren charmedBy;
+    public int charmTime; // FIXME :: when is this ever increased
+    public boolean isCharmed;
+
+    private @Nullable UUID charmedByUUID;
+    private int charmedById;
+    private boolean isInitialized;
+    private boolean triggerClientUpdate;
+
+    public void tickCharmed(final LivingEntity holder) {
+        if (!(holder instanceof Player || holder instanceof AbstractVillager || holder instanceof IHearsSiren)) {
+            return;
+        }
+
+        if (!isInitialized) {
+            initialize(holder.getLevel());
+        }
+
+        if (charmedBy == null) {
+            return;
+        }
+
+        if (charmedBy.isActuallySinging()) {
+            if (EntitySiren.isWearingEarplugs(holder) || charmTime > IafConfig.sirenMaxSingTime) {
+                charmedBy.singCooldown = IafConfig.sirenTimeBetweenSongs;
+                clearCharm();
+                return;
+            }
+
+            if (!charmedBy.isAlive() || holder.distanceTo(charmedBy) > EntitySiren.SEARCH_RANGE * 2 || holder instanceof Player player && (player.isCreative() || player.isSpectator())) {
+                clearCharm();
+                return;
+            }
+
+            if (holder.distanceTo(charmedBy) < 5) {
+                charmedBy.singCooldown = IafConfig.sirenTimeBetweenSongs;
+                charmedBy.setSinging(false);
+                charmedBy.setTarget(holder);
+                charmedBy.setAggressive(true);
+                charmedBy.triggerOtherSirens(holder);
+                clearCharm();
+                return;
+            }
+
+            isCharmed = true;
+
+            if (holder.getRandom().nextInt(7) == 0) {
+                for (int i = 0; i < 5; i++) {
+                    holder.level.addParticle(ParticleTypes.HEART,
+                            holder.getX() + ((holder.getRandom().nextDouble() - 0.5D) * 3),
+                            holder.getY() + ((holder.getRandom().nextDouble() - 0.5D) * 3),
+                            holder.getZ() + ((holder.getRandom().nextDouble() - 0.5D) * 3),
+                            0, 0, 0);
+                }
+            }
+
+            if (holder.horizontalCollision) {
+                holder.setJumping(true);
+            }
+
+            double motionXAdd = (Math.signum(charmedBy.getX() - holder.getX()) * 0.5D - holder.getDeltaMovement().x) * 0.100000000372529;
+            double motionYAdd = (Math.signum(charmedBy.getY() - holder.getY() + 1) * 0.5D - holder.getDeltaMovement().y) * 0.100000000372529;
+            double motionZAdd = (Math.signum(charmedBy.getZ() - holder.getZ()) * 0.5D - holder.getDeltaMovement().z) * 0.100000000372529;
+
+            holder.setDeltaMovement(holder.getDeltaMovement().add(motionXAdd, motionYAdd, motionZAdd));
```

Comment:
Not sure why but this doesn't seem to move the player

Reply:
tested with the fix, works again